src/Admin/BaseBlockAdmin.php

```diff
@@ -85,7 +85,14 @@ public function getNewInstance()
      */
     public function preUpdate($object)
     {
-        $this->blockManager->get($object)->preUpdate($object);
+        if (method_exists($object, 'preUpdate')) {
+            $this->blockManager->get($object)->preUpdate($object);
+
+            @trigger_error(
+                'The '.__METHOD__.' method is deprecated since version 3.x and will be removed in sonata/block-bundle 4.0.',
```

Comment:
```suggestion
                'The '.__METHOD__.'() method is deprecated since sonata-project/page-bundle 3.x and will be removed in version 4.0.',
```